src/plugins/intel_npu/src/plugin/npuw/base_sync_infer_request.cpp

```diff
@@ -196,6 +197,23 @@ void ov::npuw::IBaseInferRequest::handle_set_remote_input(const ov::Output<const
                 // ZeroRemoteTensor and ZeroHostTensor should guarantee the correct memory allocation
                 m_input_allocated.insert(tensor->data());
             }
+            // We could get a sliced RemoteTensor. In this case need to check that tensor data is allocated on the
+            // device
+            if (m_npuw_model->global_mem_device() == "NPU") {
+                auto remote_ctx =
+                    m_npuw_model->get_plugin()->get_core()->get_default_context(m_npuw_model->global_mem_device())._ptr;
+                auto zrh = remote_ctx->get_property().at(ov::intel_npu::l0_context.name());
+                if (::intel_npu::zeroUtils::memory_was_allocated_in_the_same_l0_context(
+                        static_cast<ze_context_handle_t>(zrh.as<void*>()),
+                        tensor->data())) {
+                    if (tensor->is_continuous()) {
+                        m_input_allocated.insert(tensor->data());
+                    } else {
+                        LOG_WARN("Strided remote tensor is not supported on the device! Expect worse performance due "
```

Comment:
ok in case of strided tensor -  shouldn't we clear m_input_allocated - so undo what we did in line 198?